test/integration_test_wrapper.py

```diff
@@ -0,0 +1,274 @@
+        run_mkosi(mkosi_args + ['qemu'],
+                  check=True, stderr=STDOUT, stdout=tee.stdin,
+                  **({'run': hook.wrap_run}
+                     if hook is not None and hasattr(hook, 'wrap_run')
+                     else {}))
+        tee.stdin.close()
+        try:
+            tee.wait(timeout=3)
+        except TimeoutExpired as e:
+            exit(f"Tee timeout expired: {e}")
+
+        result = run_mkosi(['--json', 'summary'],
+                           capture_output=True,
+                           check=True)
+        summary = json_loads(result.stdout)
+        image_path = [f"{image['OutputDirectory']}/{image['Output']}"
+                      for image in summary['Images']
+                      if image['Image'] == args.mkosi_image_name][0]
+        log.debug(f"Discovered disk image path {image_path}")
+
+        mounted_disk = dissect_mount
+        if hook is not None and hasattr(hook, 'mounted_disk'):
+            log.debug("Using hook.mounted_disk")
+            mounted_disk = hook.mounted_disk
+
+        try:
+            with mounted_disk(image_path) as mountpoint:
+                mountpoint = Path(mountpoint)
+
+                if hook is not None and hasattr(hook, 'check_result'):
+                    log.debug("Running check_result hook")
+                    console_log.seek(0)
+                    hook.check_result(mountpoint, console_log)
+
+                if (failed := mountpoint / "failed").exists():
+
+                    # Deliberately not handling read and decode errors
+                    # as tests should only create a regular text file
+                    msg = failed.read_text()
+
+                    if msg:
+                        print(f"Test failed: {msg}")
+                        log.debug("Test failed, non-empty /failed present")
+                        exit(1)
+
+                if (skipped := mountpoint / "skipped").exists():
+
+                    msg = ""
+                    with suppress(IOError, UnicodeDecodeError):
+                        msg = skipped.read_text()
+
+                    if msg:
+                        print(f"Test skipped: {msg}")
+
+                    log.debug("Test skipped, /skipped present")
+                    exit(77)
+
+                elif not (mountpoint / "testok").exists():
+
+                    log.debug("Test failed, /testok not present")
+                    exit(1)
+
+        except (BaseException, SystemExit) as e:
+
+            if not isinstance(e, SystemExit) or e.code != 77:
+
+                log.debug("Attempting journalctl to discover test failure")
+                run_mkosi(['journalctl',
+                           '--boot',
+                           '--unit', args.test_unit_name,
+                          ], stdin=DEVNULL)
+                raise
+
+            log.debug("Test skipped, cleaning up")
+            rmtree(per_test_output_dir)
+            raise
```

Comment:
This should be replaced with a unit drop-in credential for the test unit that we're about to run containing the following:

```
[Unit]
SuccessAction=exit
FailureAction=exit
```

That will cause the testsuite unit to stop the machine when it exits and mkosi will exit with the exit code of the unit.

Use the new `ForwardJournal=` setting in mkosi to extract the journal of the machine into a journal file on the host.

This might not be sufficient for the initramfs test yet but we'll figure that out later.

The idea is that we only need two things from the test run:

1. exit code (communicated as mkosi's exit code)
2. it's journal, extracted with `ForwardJournal=`

This means we don't need to mount the image after mkosi exits, which means we don't need to do the copy in this script, and which means we don't need to know the output dir and output image name, all of which will simplify things tremendously.